cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -72,6 +72,132 @@ void CallProfiler::print_profiling_result(const RadixTreeNode<CallerProfile> & n
+    outstream << R"("unique_id": )" << profiler.radix_tree().get_unique_node();
+    outstream << R"(})";
+}
+void CallProfilerSerializer::serialize_radix_tree_nodes(const RadixTreeNode<CallerProfile> * node, std::ostream & outstream)
+{
+    std::queue<const RadixTreeNode<CallerProfile> *> nodes_buffer;
+    nodes_buffer.push(node);
+    bool is_first_node = true;
+
+    // Serialize the nodes in a breadth-first manner.
+    while (!nodes_buffer.empty())
+    {
+        const int nodes_buffer_size = nodes_buffer.size();
+        for (int i = 0; i < nodes_buffer_size; i++)
+        {
+            const RadixTreeNode<CallerProfile> * current_node = nodes_buffer.front();
+            nodes_buffer.pop();
+            CallProfilerSerializer::serialize_radix_tree_node(*current_node, is_first_node, outstream);
+            is_first_node = false;
+            for (const auto & child : current_node->children())
+            {
+                nodes_buffer.push(child.get());
+            }
+        }
+    }
+}
+
+void CallProfilerSerializer::serialize_radix_tree_node(const RadixTreeNode<CallerProfile> & node, bool is_first_node, std::ostream & outstream)
+{
+    // Example:
+    //  {
+    //      "key":-1,
+    //      "name":"",
+    //      "data":{"start_time": 0,"caller_name": "","total_time": 0,"call_count": 0,"is_running": 0},
+    //      "children":[0]
+    //  }
+
+    // Avoid the trailing comma.
+    if (!is_first_node)
+    {
+        outstream << R"(,)";
+    }
+    outstream << R"({)";
+    outstream << R"("key": )" << node.key() << R"(,)";
+    outstream << R"("name": ")" << node.name() << R"(",)";
+    outstream << R"("data": )";
+    CallProfilerSerializer::serialize_caller_profile(node.data(), outstream);
+    outstream << R"(,)";
+    outstream << R"("children": [)";
+    bool is_first_child = true;
+    for (const auto & child : node.children())
+    {
+        // Avoid the trailing comma.
+        if (!is_first_child)
+        {
+            outstream << R"(,)";
+        }
+        is_first_child = false;
+        outstream << child->key();
+    }
+    outstream << R"(])";
+    outstream << R"(})";
+}
+
+void CallProfilerSerializer::serialize_caller_profile(const CallerProfile & profile, std::ostream & outstream)
+{
+    // Example:
+    //  {
+    //      "start_time": 0,
+    //      "caller_name": "",
+    //      "total_time": 0,
+    //      "call_count": 0,
+    //      "is_running": 0
+    //  }
+
+    outstream << R"({)";
+    outstream << R"("start_time": )" << profile.start_time.time_since_epoch().count() << R"(,)";
+    outstream << R"("caller_name": ")" << profile.caller_name << R"(",)";
+    outstream << R"("total_time": )" << profile.total_time.count() << R"(,)";
+    outstream << R"("call_count": )" << profile.call_count << R"(,)";
+    outstream << R"("is_running": )" << profile.is_running;
+    outstream << R"(})";
+}
+
 } /* end namespace modmesh */
```

Comment:
I add the comment about the serialization format.
Furthermore, I traverse the radix tree by bfs with a queue.

Reply:
It's good to use a loop to traverse. Is there a performance concern?